### PR TITLE
Add rabbitmq-c/0.10.0 & rabbitmq-c/0.9.0

### PR DESCRIPTION
**Pull request by sam-mosleh**
Specify library name and version:  **rabbitmq-c/0.10.0** and  **rabbitmq-c/0.9.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/rabbitmq-c/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    def package_info(self):
+        if self.settings.os == "Windows":
+            self.cpp_info.libs = [
+                "rabbitmq.4" if self.options.shared else "librabbitmq.4"
+            ]
+            self.cpp_info.system_libs.extend(["crypt32", "ws2_32"])
+        else:
+            self.cpp_info.libs = ["rabbitmq"]
+            self.cpp_info.system_libs.append("pthread")
```
Comment (SSE4) [inline recipes/rabbitmq-c/all/conanfile.py:81]:
```suggestion
            if self.settings.os == "Linux":
                self.cpp_info.system_libs.append("pthread")
```

Comment (sam-mosleh) [reply recipes/rabbitmq-c/all/conanfile.py:81]: Thanks for your review @SSE4. How is it going to change the behaviour?
There's an if clause for checking the OS before that.

Comment (SSE4) [reply recipes/rabbitmq-c/all/conanfile.py]: the check was `if self.settings.os == "Windows":`, so `pthread` was used for every OS. it's not needed .e.g. for Android, only for Linux

Comment (sam-mosleh) [reply recipes/rabbitmq-c/all/conanfile.py]: Good point :+1:.

**Pull request merged**